generic_config_updater/generic_updater.py

```diff
@@ -222,59 +244,139 @@ def delete_checkpoint(self, checkpoint_name):
         self.logger.log_notice(f"Checkpoint name: {checkpoint_name}.")
 
         self.logger.log_notice(f"Checking checkpoint exists.")
-        if not self._check_checkpoint_exists(checkpoint_name):
+        if not self.util.check_checkpoint_exists(checkpoint_name):
             raise ValueError(f"Checkpoint '{checkpoint_name}' does not exist")
 
         self.logger.log_notice(f"Deleting checkpoint.")
-        self._delete_checkpoint(checkpoint_name)
+        self.util.delete_checkpoint(checkpoint_name)
 
         self.logger.log_notice("Deleting checkpoint completed.")
 
-    def _ensure_checkpoints_dir_exists(self):
+
+class MultiASICConfigReplacer(ConfigReplacer):
+    def __init__(self,
+                 patch_applier=None,
+                 config_wrapper=None,
+                 patch_wrapper=None,
+                 scope=multi_asic.DEFAULT_NAMESPACE):
+        self.logger = genericUpdaterLogging.get_logger(title="MultiASICConfigReplacer",
+                                                       print_all_to_console=True)
+        self.scopelist = [HOST_NAMESPACE, *multi_asic.get_namespace_list()]
+        super().__init__(patch_applier, config_wrapper, patch_wrapper, scope)
+
+    def replace(self, target_config):
+        config_keys = set(target_config.keys())
+        missing_scopes = set(self.scopelist) - config_keys
+        if missing_scopes:
+            raise GenericConfigUpdaterError(f"To be replace config is missing scope: {missing_scopes}")
+
+        for scope in self.scopelist:
+            scope_config = target_config.pop(scope)
+            if scope.lower() == HOST_NAMESPACE:
```

Comment:
Is there any case the scope is uppercase?

Reply:
Most likely, no, but since this is going to compare with a constant, that we better align it.